Mathlib/Util/SleepHeartbeats.lean

```diff
@@ -0,0 +1,33 @@
+/-
+Copyright (c) 2023 Alex J. Best. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex J. Best
+-/
+import Lean
+
+/-!
+# Defines `sleep_heartbeats` tactic.
+
+This is useful for testing / debugging long running commands or elaboration in a somewhat precise
+manner.
+-/
+open Lean Elab
+
+/-- A low level command to sleep for at least a given number of heartbeats by running in a loop
+  until the desired number of heartbeats is hit. -/
+def sleepAtLeastHeartbeats (n : Nat) : IO Unit := do
+  let i ← IO.getNumHeartbeats
+  while (← IO.getNumHeartbeats) < i + n do
+    continue
```

Comment:
Yes I agree. I was a bit worried this function would be sufficiently inlined that it wouldn't eat heartbeats at all, but it fortunately does. I'll add a docstring note to avoid using it for anything other than testing/demoing lean meta stuff.